spring-web/src/main/java/org/springframework/http/client/MultipartBodyBuilder.java

```diff
@@ -100,6 +124,10 @@ public PartBuilder part(String name, Object part, @Nullable MediaType contentTyp
 		Object partBody;
 		HttpHeaders partHeaders = new HttpHeaders();
 
+		if (filename != null) {
```

Comment:
You should also check that filename has length.
if (StringUtils.hasLength(filename))

Reply:
Right, fixing in my next change.